src/lib/EventListener/InteractiveLoginListener.php

```diff
@@ -12,20 +12,31 @@
 namespace Novactive\EzLdapAuthenticator\EventListener;
 
 use eZ\Publish\API\Repository\UserService;
+use eZ\Publish\API\Repository\Repository;
 use eZ\Publish\Core\MVC\Symfony\Event\InteractiveLoginEvent;
 use eZ\Publish\Core\MVC\Symfony\MVCEvents;
+use \eZ\Publish\API\Repository\Values\ContentType\ContentType;
+use Symfony\Component\DependencyInjection\ContainerInterface;
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Ldap\Ldap;
 
 class InteractiveLoginListener implements EventSubscriberInterface
 {
-    /**
-     * @var \eZ\Publish\API\Repository\UserService
-     */
-    private $userService;
+    /** @var \eZ\Publish\API\Repository\Repository */
+    private $repository;
 
-    public function __construct(UserService $userService)
+    /** @var  Symfony\Component\Ldap\Ldap*/
+    private $ldap;
+
+    /** @var ContainerInterface */
+    private $container;
+
+    public function __construct(ContainerInterface $container, Repository $repository, Ldap $ldap)
```

Comment:
Please don't inject the container 🙏 , it's a bad practice...
I see that you're using it below to get parameters, then just inject a hash of options, validated by the OptionsResolver.

Reply:
Can you explain this part?

Reply:
I mean.. Yes, it's not a good idea to inject entire container. But what do you mean "hash of options, validated by the OptionsResolver"?

Reply:
hash => Associative array.

As for the OptionsResolver, it allows to validate any hash of options (define required and/or default values, types...). See http://symfony.com/doc/current/components/options_resolver.html

Reply:
Hm. Okay, but I'm not sure if it's necessary to validate it because in this case the options will be the part of the service config

Reply:
It's always useful to validate. Yes it'll be part of the service config, but this is the kind of things you might expose to user configuration. Always be defensive 😉

Reply:
defensive programming, aka always be paranoid about what user might do, aka better safe then sorry  ;)

At least when you make code for others to reuse, if this however was pure project code it probably would not be worth the time spent on it.